ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Storage/ChainDB/Impl/Args.hs

```diff
@@ -74,6 +76,12 @@ data ChainDbArgs f m blk = ChainDbArgs {
       -- is the maximum number of blocks that could be kept in memory at the
       -- same time when the background thread processing the blocks can't keep
       -- up.
+
+      -- Limit on Eagerness
+    , cdbLoELimit               :: LoELimit
+      -- ^ How many blocks can be selected beyond the LoE. The non-degenerate
+      -- value for this is @k@, the security parameter.
+    , cdbUpdateLoE              :: HKD f (UpdateLoE m blk)
```

Comment:
So my plan would be to make this the ultimate interface for the GDDG, in a synchronous fashion.

The concern you raised was, iirc, that this would mean that peers wouldn't be disconnected promptly, but only "on demand", i.e. when we want to apply the LoE limit.

Another possible issue one might suspect is that this will not be triggered after refusing to add a block, though I assume that that's not the case – can you confirm that?

At first I thought you wanted this to only be exposed to tests and not prod, but I guess you assumed that this approach was just temporary.

---

Since the limit is used directly by ChainSel while `UpdateLoE` is a black box, I think it's necessary to have both as parameters.
However, the limit is basically a boolean and we use `k` for the value unconditionally, so we could theoretically merge them so that `UpdateLoE` has a constructor that indicates it's disabled, and ChainSel would match on the callback constructor to decide to apply the limit...probably bad style though.

Reply:
> Another possible issue one might suspect is that this will not be triggered after refusing to add a block, though I assume that that's not the case – can you confirm that?

By "refusing to add a block" you mean the event that we don't end up selecting the block? I think `updateLoE` is called sufficiently early (ie before checks whether it already is in the VolDB) that it should not be skipped even in that case.

---

Overall, the impact of `updateLoEFrag` is very minimal, and it is not a required part of the LoE deliverable, so we can leave it for now and decide this later when we implement the GDD.